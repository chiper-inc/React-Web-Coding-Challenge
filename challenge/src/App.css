.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
#root, html, body, #root, .root {
  height: 100%;
}
.container-bike-form {
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
}
.login-form {
  padding: 20px;
  border: 1px lightgrey solid;
  border-radius: 10px;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  min-width: 300px;
  min-height: 250px;
}
.login-logo {
  width: 100px;
}
.login-button {
  width: 100%;
}
.header-bikes {
  width: 100%;
  background: #222222;
  padding: 10px 50px;
  display: flex;
  border-bottom-left-radius: 10px;
  border-bottom-right-radius: 10px;
  color: white;
  margin-bottom: 20px;
}
.header-bikes .bike-logo-nav {
  width: 100px;
  margin-right: 20px;
}
.list-bikes {
  width: 100%;
  padding: 10px;
  display: flex;
}
.image-list-bake {
  width: 200px;
  height: 200px;
  border: 1px solid black;
  margin-right: 10px;
}
.image-list-bake img {
  width: 100%;
  height: 100%;
  background-size: 100% 100%;
}
.detail-list-bike {
  display: flex;
  width: calc((100% - 200px) / 2);
  flex-direction: column;
}
.list-bikes-container {
  margin-top: 20px;
}
.detail-list-bike h3 {
  color: #3498db;
}
.content-dashboard {
  padding-left: 30px;
  padding-right: 30px;
}
.spinner-container {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 50px;
}
.pagination-container {
  width: 100%;
  padding: 20px;
}
