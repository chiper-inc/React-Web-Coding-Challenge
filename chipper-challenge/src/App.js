import React from 'react';
import { Route } from 'react-router-dom';
import Home from './components/Home/Home';
import NavBar from './components/NavBar/NavBar';
import Scroll from './components/Scroll/Scroll';

function App() {
    return (
        <>
            <Route path='/' component={NavBar} />
            <Route path='/' component={Home} />
            <Route path='/' component={Scroll} />
        </>
    );
}

export default App;
