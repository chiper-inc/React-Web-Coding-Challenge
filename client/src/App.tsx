import React from "react";
import { Route } from 'react-router-dom';
import NavBar from "./components/NavBar/NavBar";
import Filters from './components/Filters/Filters'

const App: React.FC = () => {
  return (
    <div className="App">
      <Route path='/' component={NavBar} />
      <Route path='/' component={Filters} />
      <h1>Hello World!</h1>
    </div>
  );
};

export default App;