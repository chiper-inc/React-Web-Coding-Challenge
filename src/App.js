import React from 'react'
import { Switch, Route } from 'react-router-dom'
import { HOME, MAP } from './routes'
import { Layout } from './components/Layout'
import { Home } from './entries/Home'
import { Map } from './entries/Map'
import { NotFound } from './entries/NotFound'
import { AppStyles } from './AppStyles'

const App = () => (
  <Layout>
    <AppStyles />
    <Switch>
      <Route exact path={HOME} component={Home} />
      <Route exact path={MAP} component={Map} />
      <Route path='*' component={NotFound} />
    </Switch>
  </Layout>
)

export default App
