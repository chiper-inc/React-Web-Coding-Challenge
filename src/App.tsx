// import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { Header } from './components';
import { Details, Home } from './views';

function App() {
  return (
    <>
      <Header />
      <div className="container mx-auto px-4 mt-8">
        <Router>
          <Switch>
            <Route path="/details/:id">
              <Details />
            </Route>
            <Route path="/">
              <Home />
            </Route>
          </Switch>
        </Router>
      </div>
    </>
  );
}

export default App;
