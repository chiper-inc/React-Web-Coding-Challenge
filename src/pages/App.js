import React from "react";
import Bike from "../context/bike";

export default () => <div><Bike /></div>
