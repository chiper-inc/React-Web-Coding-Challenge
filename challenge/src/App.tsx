import React from 'react';
import './App.css';
import Router from './router';
import 'bootstrap/dist/css/bootstrap.min.css';

const App = () => {
  return (
    <div>
      <Router />
    </div>
  );
};

export default App;
