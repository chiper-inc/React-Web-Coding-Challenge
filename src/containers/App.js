import React from "react";
import { Route, Switch } from "wouter";
import { Grid } from "@material-ui/core";
import { Header } from "../components/Header";
import { Detail } from "../pages/Detail";
import { ErrorPage } from "../pages/ErrorPage";
import { Home } from "../pages/Home";
import { IncidentsContextProvider } from "../context/IncidentsContext";

export const App = () => {
  return (
    <>
      <Grid container spacing={2}>
        <Grid item xs={12}>
          <Header />
        </Grid>
      </Grid>
      <IncidentsContextProvider>
        <Switch>
          <Route component={Home} path="/" />
          <Route component={Detail} path="/detail/:id" />
          <Route component={ErrorPage} path="/:rest*" />
        </Switch>
      </IncidentsContextProvider>
    </>
  );
};
