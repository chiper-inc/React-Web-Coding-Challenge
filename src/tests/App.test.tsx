import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';
test('renders the component', () => {
  const component = shallow(<App />);
  expect(component).toMatchSnapshot();
});
