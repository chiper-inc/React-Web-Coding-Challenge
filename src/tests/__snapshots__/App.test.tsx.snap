// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component App 1`] = `
<Fragment>
  <Header />
  <div
    className="container mx-auto px-4 mt-8"
  >
    <BrowserRouter>
      <Switch>
        <Route
          path="/details/:id"
        >
          <Details />
        </Route>
        <Route
          path="/"
        >
          <Home />
        </Route>
      </Switch>
    </BrowserRouter>
  </div>
</Fragment>
`;
